File: python-impl/insfrastructure/usecases/delete_account_usecase.py
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class DeleteAccountUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, account_id: int) -> bool:
        return self.bank_account_repo.delete(account_id)
        

File: typescript-impl/src/infrastructure/usecases/delete_account_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';

export class DeleteAccountUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(account_id: number): boolean {
        return this.bank_account_repo.delete(account_id);
    }
}
